admin/class-primary-term-admin.php

```diff
@@ -0,0 +1,195 @@
+
+		// Only enqueue if there are taxonomies that need a primary term.
+		if ( empty( $taxonomies ) ) {
+			return;
+		}
+
+		wp_enqueue_style( 'wpseo-primary-category', plugins_url( 'css/metabox-primary-category' . WPSEO_CSSJS_SUFFIX . '.css', WPSEO_FILE ), array(), WPSEO_VERSION );
+
+		wp_enqueue_script( 'wpseo-primary-category', plugins_url( 'js/wp-seo-metabox-category' . WPSEO_CSSJS_SUFFIX . '.js', WPSEO_FILE ), array( 'jquery' ), WPSEO_VERSION, true );
+
+		$taxonomies = array_map( array( $this, 'map_taxonomies_for_js' ), $taxonomies );
+
+		$data = array(
+			'taxonomies' => $taxonomies,
+		);
+		wp_localize_script( 'wpseo-primary-category', 'wpseoPrimaryCategoryL10n', $data );
+	}
+
+	/**
+	 * Saves all selected primary terms
+	 *
+	 * @param int $post_ID
+	 */
+	public function save_primary_terms( $post_ID ) {
+		$taxonomies = $this->get_primary_term_taxonomies( $post_ID );
+
+		foreach ( $taxonomies as $taxonomy ) {
+			$this->save_primary_term( $post_ID, $taxonomy );
+		}
+	}
+
+	/**
+	 * Filters post_link_category to change the category to the chosen category by the user
+	 *
+	 * @param stdClass $category The category that is now used for the post link.
+	 *
+	 * @return array|null|object|WP_Error The category we want to use for the post link.
+	 */
+	public function post_link_category( $category ) {
+		$primary_category = new WPSEO_Primary_Term( 'category', get_the_ID() );
+		$primary_category = $primary_category->get_primary_term();
+
+		if ( false !== $primary_category && $primary_category !== $category->cat_ID ) {
+			$category = get_category( $primary_category );
+		}
+
+		return $category;
+	}
+
+	/**
+	 * Returns an array suitable for use in the javascript
+	 *
+	 * @param stdClass $taxonomy
+	 * @return array
+	 */
+	private function map_taxonomies_for_js( $taxonomy ) {
+		$terms = get_terms( $taxonomy->name );
+
+		$primary_term = new WPSEO_Primary_Term( $taxonomy->name, get_the_ID() );
+
+		return array(
+			'title'   => $taxonomy->labels->singular_name,
+			'name'    => $taxonomy->name,
+			'primary' => $primary_term->get_primary_term(),
+			'terms'   => array_map( array( $this, 'map_terms_for_js' ), $terms ),
+		);
+	}
+
+	/**
+	 * Returns an array suitable for use in the javascript
+	 *
+	 * @param stdClass $term
+	 *
+	 * @return array
+	 */
+	private function map_terms_for_js( $term ) {
+		return array(
+			'id'   => $term->term_id,
+			'name' => $term->name,
+		);
+	}
+
+	/**
+	 * Save the primary term for a specific taxonomy
+	 *
+	 * @param int      $post_ID
+	 * @param stdClass $taxonomy
+	 */
+	private function save_primary_term( $post_ID, $taxonomy ) {
+		$primary_term = filter_input( INPUT_POST, WPSEO_Meta::$form_prefix . 'primary_' . $taxonomy->name . '_term', FILTER_SANITIZE_NUMBER_INT );
+
+		if ( null === $primary_term ) {
+			return;
+		}
+
+		if ( ! check_admin_referer( 'save-primary-term', WPSEO_Meta::$form_prefix . 'primary_' . $taxonomy->name . '_nonce' ) ) {
```

Comment:
Can you combine the two if-statements because of the result is the same and it will lower the complexity of this method.